File: chap08/run-model8-6.R
library(cmdstanr)

d <- read.csv('input/data-conc-2.csv')
N <- nrow(d)
X <- c(1, 2, 4, 8, 12, 24)
T <- length(X)
Tp <- 60
Xp <- seq(from=0, to=24, length=Tp)
data <- list(N=N, T=T, X=X, Y=d[,-1], Tp=Tp, Xp=Xp)

model <- cmdstan_model(stan_file='model/model8-6.stan')
fit <- model$sample(data=data, seed=123, parallel_chains=4)
fit$save_object(file='output/result-model8-6.RDS')

File: chap08/run-model8-6.py
import pandas
import numpy as np
import cmdstanpy

d = pandas.read_csv('input/data-conc-2.csv')
N = len(d)
X = [1, 2, 4, 8, 12, 24]
T = len(X)
Tp = 60
Xp = np.linspace(0, 24, num=Tp)
data = {'N':N, 'T':T, 'X':X, 'Y':d.drop('PersonID', axis=1), 'Tp':Tp, 'Xp':Xp}

model = cmdstanpy.CmdStanModel(stan_file='model/model8-6.stan')
fit = model.sample(data=data, seed=123, parallel_chains=4)
fit.save_csvfiles('output/result-model8-6')
